File: snippets/bintoc.rb
#!/usr/bin/ruby

# Check if both input and output filenames were provided as arguments
if ARGV.length < 2
  puts "Usage: #{$0} <input_file> <output_file>"
  exit 1
end

input_filename  = ARGV[0]
output_filename = ARGV[1]

# Originally Cod3d By 0xNinjaCyclone, command line arguments and file output added by @toneillcodes
# https://github.com/0xNinjaCyclone/EarlyCascade/blob/main/bintoc.rb
File.open(input_filename, "rb") do |input_file|
  File.open(output_filename, "w") do |output_file|
    output_file.print "unsigned char buf[] =    "
    while buffer = input_file.read(16)
      output_file.print "\n#{' ' * 4 * 5}\""
      buffer.bytes.each do |byte|
        output_file.print "\\x%0.2x" % byte
      end
      output_file.print '"'
    end
    output_file.puts ';'
  end
end

File: snippets/bintoc.py
#!/usr/bin/python
import sys

# Check if both input and output filenames were provided as arguments
if len(sys.argv) < 3:
    print(f"Usage: {sys.argv[0]} <input_file> <output_file>")
    sys.exit(1)

input_filename = sys.argv[1]
output_filename = sys.argv[2]

# Originally Cod3d By 0xNinjaCyclone, ported to Python with input/output command line arguments added by @toneillcodes
# https://github.com/0xNinjaCyclone/EarlyCascade/blob/main/bintoc.rb
with open(input_filename, "rb") as input_file:
    with open(output_filename, "w") as output_file:
        output_file.write("unsigned char buf[] =    ")
        while True:
            buffer = input_file.read(16)
            if not buffer:
                break
            output_file.write(f"\n{' ' * 20}\"")
            for byte in buffer:
                output_file.write(f"\\x{byte:02x}")
            output_file.write('"')
        output_file.write(';')